File: src/go/cat.go
package main

import ("fmt"
	"io"
	"os")

func main() {
	f, err := os.Open("../data")
	if err != nil {
		fmt.Println(err)
		return
	}
	defer f.Close()
	data := make([]byte, 131072)
	for {
		data = data[:cap(data)]
		n, err := f.Read(data)
		if err != nil {
			if err == io.EOF {
				break
			}
			fmt.Println(err)
			return
		}
		data = data[:n]
		os.Stdout.Write(data)
	}
}

File: src/nodejs/cat.js
#!/usr/bin/nodejs

var fs = require('fs');
var process = require('process');

fs.open('../data','r',
	function(status,fd){
	    if (status) {
		console.log(status.message);
		return;
	    }
	    var buffer = new Buffer(131072);
	    var eof=false;
	    while (! eof) {
		fs.read(fd, buffer, 0, 131072, 0, function(err, num, buffer) {
		    if (num == 0) { eof=true; }
		    else {
			process.stdout.write(buffer);
		    }
		});
	    }
	    process.exit(0);
	});
